### ISSUE TITLE
[BUG] Get-PnPExternalUser fails with message "SyntexFeatureLevelOnOff flight is not enabled"

### ISSUE THREAD

**Issue by pmatthews05** [pnp/powershell#3353]
### Reporting an Issue or Missing Feature
When calling Get-PNPExternalUser, it fails with the following error

`SyntextFeatureLevelOnOff flight is not enabled`

This was working 3 days ago, and stop working this morning. I've tested on a couple of tenants.

### Expected behavior 
Expected to see a list of external users.

### Actual behavior 
Fails with error message: "SyntextFeatureLevelOnOff flight is not enabled"

![image](https://github.com/pnp/powershell/assets/6736509/e795beef-b8ca-4714-ac95-f5d6aca652fa)

### Steps to reproduce behavior
```ps1
Connect-PnpOnline -Url:"https://<sharepoint>-admin.sharepoint.com -PnPManagementShell
Get-PnPExternalUser
```

### What is the version of the Cmdlet module you are running?
2.2.0

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [x] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by rdicicco** [comment]
I am getting this same error message when using the Disable-PnPSharingForNonOwnersOfSite command as well. 
![image](https://github.com/pnp/powershell/assets/49653021/45056cea-4032-4bda-bed9-1def230baedf)

**Comment by andreyp115** [comment]
Same error, started today:
![image](https://github.com/pnp/powershell/assets/142851557/8a8f4cbe-2997-4895-83b7-e37005f94728)

**Comment by sabadraprasham** [comment]
Facing same error in our tenant. 

But its working in local development tenant. 

Please suggest how to solve this issue

**Comment by rajmcablr** [comment]
> Facing same error in our tenant.
> 
> But its working in local development tenant.
> 
> Please suggest how to solve this issue

We are also experiencing the same issue.
Can someone from MS guide us to resolve it or any workaround to would work too?
pls. guide ASAP

**Comment by joaquinajenjo** [comment]
Same issue with the SharePoint PowerShell command:
`Get-SPOExternalUser`

**Comment by NorbertPU** [comment]
This problem starts with creating a new TeamSite with the following command:
`New-PnPSite -Type TeamSite -Title ($TeamName) -Alias $TeamSPOName -Description $ProjectName -Owners $ObjectOwner -Wait`

This currently results in an SPO site without the sharing options at the permissions section and most of the site settings.
Subsequently other commands like below fail with the SyntexFeatureLevelOnOff message.
`Set-SPOSite -Identity $SiteCollURL -DisableSharingForNonOwners`

![image](https://github.com/pnp/powershell/assets/142881045/085b385a-380e-4a17-9e36-39a78c4ec43f)

![image](https://github.com/pnp/powershell/assets/142881045/4ada5a8a-b313-4e01-b92d-72494b2dbb91)

**Comment by Sharepoint2k22** [comment]
Same issue with:
Set-PnPSite <SPSiteUrl> -DisableSharingForNonOwners

**Comment by enthusol** [comment]
Same issue with `Set-PnPSite -Sharing ...` (fails in 2.2.0 and also 1.3.0).

**Comment by FlorianKornmueller** [comment]
same here.... 

` LogS "General page configuration"

        #Region How to share Setting        
        LogC "Set Sharing to 'Allow sharing only for owners'..."
        Set-PnPSite -Identity $site.Url -DisableSharingForNonOwners -Connection $connection -ErrorAction SilentlyContinue -ErrorVariable ErrPnpSiteShareing | Out-null        

        if($ErrPnpSiteShareing){
            Exit-WithError -exitCode 6 -message $ErrPnpSiteShareing     
            #LogE  ("   " + $ErrPnpSiteShareing)
        }`
fails with    -    SyntexFeatureLevelOnOff flight is not enabled

**Comment by Nicomadreja87** [comment]
Same here!, not only with the PNP module but also with Microsoft.Online.SharePoint.PowerShell

**Comment by NishkalankBezawada** [comment]
> Same here!, not only with the PNP module but also with Microsoft.Online.SharePoint.PowerShell

Found the issue, Will create a new Pull request soon. 

Thanks,
Nishkalank

**Comment by LastGunslinger** [comment]
@NishkalankBezawada, you say that you've found the issue, can you elaborate? I ask because this is not just an issue with the PnP.PowerShell module, so I'm not understanding how this can be fixed with a PR just for this module.

**Comment by NishkalankBezawada** [comment]
> @NishkalankBezawada, you say that you've found the issue, can you elaborate? I ask because this is not just an issue with the PnP.PowerShell module, so I'm not understanding how this can be fixed with a PR just for this module.

![image](https://github.com/pnp/powershell/assets/47456098/e9865867-39a9-4aab-90e3-432921a95fb3)

Line number 32 is causing the issue.

//Nish

**Comment by NishkalankBezawada** [comment]
Hello @pmatthews05 @rdicicco @andreyp115 @sabadraprasham @joaquinajenjo @Sharepoint2k22 @enthusol @FlorianKornmueller @Nicomadreja87 

I have just created a [PR 3357](https://github.com/pnp/powershell/pull/3357) which will fix the issue. 

Thanks,
Nishkalank Bezawada

**Comment by wpsgo** [comment]
Thank you Nishkalank.   Can you include a note here on how we would go about getting access to this updated module?   Should we just download the latest version of the PnPPowerShell Module?

**Comment by NishkalankBezawada** [comment]
> Thank you Nishkalank.   Can you include a note here on how we would go about getting access to this updated module?   Should we just download the latest version of the PnPPowerShell Module?   
> 
> 

Hello,

Once the PR is merged into the Dev branch, the updated module will be available on the latest nightly release. I'm not sure when the PR will be merged but yes I'll keep you posted here.

Thanks,
Nishkalank Bezawada

**Comment by enthusol** [comment]
@NishkalankBezawada Thank you for the quick solution, great work! Would it be possible to add a fix for PnP.PowerShell 1.12.0 as well?

We have a lot of customers that rely on PnP.PowerShell 1.x.x because the module is used on runbooks in Azure Automation Accounts with PowerShell 5.1 (because 7.1 and 7.2 are still in preview).

**Comment by NorbertPU** [comment]
From my point of view the problem SyntexFeatureLevelOnOff-Problem starts a bit earlier with **New-PnPSite**
Sites created through New-PnPSite (PnP.Powershell 1.12.0) are not fully functional:
The settings https://TENANT.sharepoint.com/sites/SITE/_layouts/15/settings.aspx show only part of the settings:
![image](https://github.com/pnp/powershell/assets/142881045/ab68b4ee-afd5-488f-9eb3-92e1ee660513)

The settings should look like this:
![image](https://github.com/pnp/powershell/assets/142881045/c3b3b549-ba12-45d3-98d2-12417f86596f)

The same problem with the sharing permissions:
They look like this after using New-PnPSite
![image](https://github.com/pnp/powershell/assets/142881045/2f72328d-533a-4334-8ccd-f7c8ba658cec)

They should look like this:
![image](https://github.com/pnp/powershell/assets/142881045/4759a290-30ad-45ae-9112-9a24ed5922ba)

When I manually create a TeamSite through the SharePoint Admin-Center I can then use commands like `Set-SPOSite -Identity $SiteCollURL -DefaultLinkToExistingAccess $true` or `Disable-PnPSharingForNonOwnersOfSite` without having the SyntexFeatureLevelOnOff-Problem.

Does anybody else experience this behaviour with New-PnPSite?

**Comment by enthusol** [comment]
@NorbertPU Now I am highly confused. I tried to reconstruct your described behaviour in several tenants but everything just looks fine.

After that I retried the earlier mentioned cmdlets (e. g. `Set-PnPSite -Sharing ...`) and everything works without problems just like last week. **Does it work again for everyone?** Maybe Microsoft made changes on their side? 🤔

**Comment by andreyp115** [comment]
Get-SPOExternalUser works for me now also.

**Comment by Sharepoint2k22** [comment]
Working again here too, this is using Set-PnPSite -Identity $Url -DisableSharingForNonOwners

**Comment by NorbertPU** [comment]
I ran my script to setup a new SPO and Team again -  `Set-SPOSite -Identity $SiteCollURL -DefaultLinkToExistingAccess $true` or `Disable-PnPSharingForNonOwnersOfSite` are working without errors now (PnP.Powershell 1.12.0)

**Comment by NishkalankBezawada** [comment]
Tested from my end too, All the commands reported to be not working are now working again. Will close the [PR 3357](https://github.com/pnp/powershell/pull/3357)


//Nishkalank

**Comment by sabadraprasham** [comment]
Get-SPOExternalUser is started working for me again :)

**Comment by Nicomadreja87** [comment]
We still have the problem. Both with the PnP commands (Get-PnPExternalUser) and with those of the sharepoint module (Get-SPOExternalUser) for what seems to be a Microsoft issue. If anyone has done any procedures to get it working again, I'd be grateful if you'd share it :)

**Comment by LastGunslinger** [comment]
@Nicomadreja87 I opened a ticket with Microsoft support and was told that they are aware of the issue and are rolling out a hotfix to affected areas by this Friday. When I tried to reproduce this issue today I found that the `Get-PnPExternalUser` command is no longer failing with the SyntexFeatureLevelOnOff error, **BUT** it is still not working normally. Instead of returning a list of external user accounts, no matter what parameters I gave to the command it returned just one _non-external_ account, so this is still very much not fixed on Microsoft's side.

**Comment by andreyp115** [comment]
I am getting expected results:
![image](https://github.com/pnp/powershell/assets/142851557/e9dea3a2-3a7c-421c-9225-e8d94143da1d)
